src/components/App/App.actions.js

```diff
@@ -114,6 +116,15 @@ export function updateUserDataFromAPI() {
       const { id } = userData;
       const newUserData = await API.getUserData(id);
       dispatch(updateUserData({ ...userData, ...newUserData }));
+
+      if (newUserData.settings?.speech?.elevenLabsApiKey) {
+        dispatch(
+          changeElevenLabsApiKey(newUserData.settings.speech.elevenLabsApiKey)
+        );
+        tts.reinitializeElevenLabs(
+          newUserData.settings.speech.elevenLabsApiKey
+        );
+      }
```

Comment:
Not sure if this should be performed here

Reply:
I see! You are right, we should update the settings after a page refresh. The problem here is this should not be performed here as this action it should perfrom only `userDate` related actions. We should fix this on the future